app/models/topic.rb

```diff
@@ -126,6 +126,11 @@ def assign(user_id=2, assigned_to)
     self.save
   end
 
+  def self.bulk_assign(post_attributes, assigned_to)
+    Post.create(post_attributes)
```

Comment:
I suspect if we could figure out how to do a true bulk insert (instead of looping through) that would shave quite a bit of time.  If not, just this action alone could be a candidate for background processing if the number of `topics` is greater than 1 (ie. we are on the list view).